lib/monzo/pot.rb

```diff
@@ -25,13 +27,89 @@ def initialize(params)
     #
     # Returns An Array of Monzo::Pot
     def self.all
-      client = Monzo.client
-      response = client.get("/pots/listV1")
+      response = Monzo.client.get("/pots")
       parsed_response = JSON.parse(response.body, :symbolize_names => true)
 
       parsed_response[:pots].map do |item|
         Monzo::Pot.new(item)
       end
     end
+
+    # Public: Find a pot with the given pot id.
+    #
+    # pot_id - The id to find.
+    #
+    # Returns an instance of Monzo::Pot.
+    def self.find(pot_id)
+      response = Monzo.client.get("/pots/#{pot_id}")
+      parsed_response = JSON.parse(response.body, :symbolize_names => true)
+
+      Monzo::Pot.new(parsed_response[:pot])
+    end
+
+    # Public: Deposit Money in a pot
+    #
+    # amount - The amount to deposit, in pennies.
+    # source_account_id - The account_id of the account to withdraw from.
+    # dedupe_id (optional) - A random string, to prevent duplicate deposits.
```

Comment:
@dannysmith I am wondering what your thoughts on whether this should be optional. From reading the Monzo docs it would appear `dedupe_id` should be:

> A unique string used to de-duplicate deposits. Ensure this remains static between retries to ensure only one deposit is created.

In your implementation a new dedupe_id is created every time `deposit!` or `withdraw!` is called which saves the user having to think about generating a random id. However it does mean a user of the client could create accidental duplicate requests on retries. I wonder if it would be better to reveal this more to the user by providing a deduplication id generator and then making the `dedupe_id` a required field. 

```
dedupe_id = Monzo::Duplication.create_id

pot.withdraw(100, account_id, dedupe_id)

# A retry with the same dedupe id
pot.withdraw(100, account_id, dedupe_id)

```

What do you think?

Reply:
I thought about this a bunch and decided against requiring it. I reckon ease of use outweighs the need for dupe protection, particularly considering the type of things people are building on the Monzo API at the moment (ie. mainly toy apps for their own use). I'd hope that anyone building something that's retrying failed transactions would be sensible enough to include `dedupe_id`.

That said, we can't expect everyone to magically know it's an available argument so I'm about to add a section to the README by way of explaination.

Let me know if you disagree with this thinking and I'll make it a required argument.

Either way, I don't think we should provide a generator as part of this gem - the job of generating random strings seems a bit outside its remit, and Ruby's `SecureRandom` does the job easily enough.

Reply:
@dannysmith Yeah everything you have said crossed my mind as well. Glad we are on the same page about the tradeoffs. 👍 

I think lets ship it as being an optional argument. If we get people complaining it is unexpected we can revisit it later.

Thanks for your thoughts and opinions. 🙌